### PR TITLE
Zoom methods

### PR DESCRIPTION
**Pull request by kmathew-pdftron**
More zoom Methods will be posted in a separate PR

### CODE REVIEW

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 1]
```diff
@@ -2395,6 +2403,19 @@ public static void onMethodCall(MethodCall call, MethodChannel.Result result, Vi
                 }
                 break;
             }
+            case FUNCTION_GET_ZOOM: {
+                checkFunctionPrecondition(component);
+                getZoom(result, component);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:2408]: break

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 2]
```diff
@@ -2395,6 +2403,19 @@ public static void onMethodCall(MethodCall call, MethodChannel.Result result, Vi
...
+            }
+            case FUNCTION_SET_ZOOM_LIMITS: {
+                checkFunctionPrecondition(component);
+                try {
+                    setZoomLimits(call, result, component);
+                } catch (PDFNetException ex) {
+                    ex.printStackTrace();
+                    result.error(Long.toString(ex.getErrorCode()), "PDFTronException Error: " + ex, null);
+                }
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:2417]: break

File: example/lib/main.dart [hunk 3]
```diff
@@ -212,6 +212,8 @@ class _ViewerState extends State<Viewer> {
     });
 
     await controller.openDocument(_document, config: config);
+
+ await controller.setZoomLimits(ZoomLimitMode.Absolute, 1.0, 3.5);
```
Comment (sgong-pdftron) [inline example/lib/main.dart:216]: i'd remove it from sample code because it's not a common use case

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java [hunk 4]
```diff
@@ -2429,6 +2452,32 @@ public static void onMethodCall(MethodCall call, MethodChannel.Result result, Vi
...
+                limitMode = PDFViewCtrl.ZoomLimitMode.ABSOLUTE;
+                break;
+            case KEY_ZOOM_LIMIT_MODE_RELATIVE:
+                limitMode = PDFViewCtrl.ZoomLimitMode.RELATIVE;
+                break;
+            case KEY_ZOOM_LIMIT_MODE_NONE:
+                limitMode = PDFViewCtrl.ZoomLimitMode.NONE;
+                break;
+        }
+        pdfViewCtrl.setZoomLimits(limitMode, minimum, maximum);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/pdftronflutter/helpers/PluginUtils.java:2478]: missing null check on limitMode, see RN code for reference

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

**Review by sgong-pdftron — APPROVED** [review]

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]
missing docs

File: ios/Classes/PdftronFlutterPlugin.m [hunk 5]
```diff
@@ -2689,6 +2693,27 @@ - (void)getCurrentPage:(FlutterResult)flutterResult {
     flutterResult([NSNumber numberWithInt:documentController.pdfViewCtrl.currentPage]);
 }
 
+- (void)getZoom:(FlutterResult)flutterResult {
+    PTDocumentController *documentController = [self getDocumentController];
+    double zoom = [documentController.pdfViewCtrl GetZoom];
```
Comment (JamieDass) [inline ios/Classes/PdftronFlutterPlugin.m:2698]: I noticed the RN version returns:
`pdfViewCtrl.zoom * pdfViewCtrl.zoomScale`
on iOS at least.
Should it be the same here?

Comment (kmathew-pdftron) [reply ios/Classes/PdftronFlutterPlugin.m:2698]: this works but I see that is a lot shorter

Comment (JamieDass) [reply ios/Classes/PdftronFlutterPlugin.m:2698]: Hm not sure what you mean by shorter?

Comment (kmathew-pdftron) [reply ios/Classes/PdftronFlutterPlugin.m:2698]: maybe not

**Review by sgong-pdftron — APPROVED** [review]

**Review by JamieDass — APPROVED** [review]

**Pull request merged**